### PR TITLE
Add trigger to disallow updating immutable table

### PR DESCRIPTION
**Pull request by edmofro**
I was thinking of doing this for `entity_relation`, but thought I would go down the proper route (see earlier PR). However, this is still very useful for ancestor_descendant_relation (and any other derived tables in future)

### CODE REVIEW

File: packages/database/src/runPostMigration.js [hunk 1]
```diff
@@ -27,37 +45,52 @@ export const runPostMigration = async driver => {
...
+        FOR EACH ROW EXECUTE PROCEDURE notification();
+      `),
+    ),
+  );
+
+  // Find any table in the database that doesn't have a corresponding notification trigger with
+  // the name {TABLE_NAME}_immutable_trigger (eg survey_response_immutable_trigger).
+  const { rows: tablesWithoutImmutableTriggerResults } = await driver.runSql(`
+    ${SELECT_TABLES_WITHOUT_TRIGGERS}
+    AND t.table_name IN (${IMMUTABLE_TABLES.join(',')});
```
Comment (rohan-bes) [inline packages/database/src/runPostMigration.js:70]: Super trivial, but perhaps it's a bit nicer to use the `arrayToDbString` function here? Saves having to remember to wrap the immutable table names in quotes

**Review by rohan-bes — APPROVED** [review]
Looks really good 😄 

Just want to confirm, so the idea is that we will make all tables without existing trigger functions to be immutable, on top of the explicitly declared immutable ones?

Just one comment but it's up to you if you want to do it 👍

File: packages/database/src/runPostMigration.js (hunk 1 above)
Comment (edmofro) [reply packages/database/src/runPostMigration.js:70]: Oh yep, that'd be a nice improvement :+1:

**Comment by edmofro** [comment]
Actually it's not the case that tables that don't have change notifiers are immutable - they're two distinct groups. But some tables we do want to make immutable - and now we have the ability to bake that in at the db error level :-)

**Comment by rohan-bes** [comment]
> Just want to confirm, so the idea is that we will make all tables without existing trigger functions to be immutable, on top of the explicitly declared immutable ones?

> Actually it's not the case that tables that don't have change notifiers are immutable - they're two distinct groups. But some tables we do want to make immutable - and now we have the ability to bake that in at the db error level :-)

Omg I misread the `AND` as an `OR` the code here: 
` ${SELECT_TABLES_WITHOUT_TRIGGERS} AND t.table_name IN (${arrayToDbString(IMMUTABLE_TABLES)});`

Sorry for the confusion 😝

**Review by rohan-bes — APPROVED** [review]
Looks good, nice catch on the bug where the immutability trigger name wasn't matching `<tableName>_trigger` 👍

**Pull request merged**